### PR TITLE
feat(Ecto.Migration.Runner): Support drop_if_exists for Ecto.Migratio…

### PR DESCRIPTION
**Pull request by tomciopp**
…n.Constaints

@josevalim Adds support for dropping constraints if they exist. 

https://www.postgresql.org/docs/9.6/sql-altertable.html

### CODE REVIEW

**Comment by josevalim** [comment]
@tomciopp can you please add a test? We need a test for logging in migration_test and we need a test for both mysql / postgres to make sure they emit the proper expressions. Thanks!

**Comment by tomciopp** [comment]
Ok, will do.

**Comment by tomciopp** [comment]
@josevalim I believe that should do the trick. Let me know if I missed anything.

File: test/ecto/migration_test.exs [hunk 1]
```diff
@@ -665,6 +671,13 @@ defmodule Ecto.MigrationTest do
     assert {:create, %Index{}} = last_command()
   end
 
+  test "backward: drops a constraint" do
+    assert_raise Ecto.MigrationError, ~r/cannot reverse migration command/, fn ->
```
Comment (josevalim) [inline test/ecto/migration_test.exs]: Is there a reason why we can't reverse it?

Comment (josevalim) [reply test/ecto/migration_test.exs]: That's a good point. Ok, let's roll as is then. :)

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Pull request merged**